### PR TITLE
Remove integration edit route

### PR DESCRIPTION
**Pull request by josemigallas**
Closes [THREESCALE-6460: Remove integrations/edit route and pages](https://issues.redhat.com/browse/THREESCALE-6460)

### CODE REVIEW

File: test/unit/presenters/api/integrations_show_presenter_test.rb [hunk 1]
```diff
@@ -50,8 +50,6 @@ def test_environments_have_same_config?
   end
 
   test 'state_modifier' do
-    @proxy.account.stubs(:provider_can_use?).with(:api_as_product).returns(true)
```
Comment (Martouta) [inline test/unit/presenters/api/integrations_show_presenter_test.rb:53]: I recommend not doing this change here to avoid conflicts with https://github.com/3scale/porta/pull/2291, where you actually remove this entire test.
But it's up to you 👍🏼

File: cdn/font-awesome/4.3.0/css/font-awesome.css [hunk 2]
```diff
@@ -913,7 +913,6 @@
 .fa-tablet:before {
   content: "\f10a";
 }
-.fa-mobile-phone:before,
```
Comment (Martouta) [inline cdn/font-awesome/4.3.0/css/font-awesome.css:916]: Then also remove it from here: https://github.com/3scale/porta/blob/master/cdn/font-awesome/4.3.0/css/font-awesome.min.css

File: app/helpers/services_helper.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@ def show_mappings?
   end
 
   def path_to_service(service)
```
Comment (Martouta) [inline app/helpers/services_helper.rb:29]: plz remove this method and use directly `admin_service_integration_path(service)` instead

Comment (Martouta) [reply app/helpers/services_helper.rb:29]: Not necessarily in this same PR 👍🏼

File: app/controllers/api/integrations_controller.rb [hunk 4]
```diff
@@ -66,15 +65,15 @@ def update_production
 
     done_step(:apicast_gateway_deployed, final_step=true) if ApiClassificationService.test(@proxy.api_backend).real_api?
 
-    redirect_to action: :edit, anchor: 'proxy'
+    redirect_to action: :show, anchor: 'proxy'
```
Comment (Martouta) [inline app/controllers/api/integrations_controller.rb:68]: The problem of redirecting here to a legit action is that then it seems like this is also a reachable legit path, and it isn't 😄 so I propose removing it directly since it is not used anymore (for APIAP) 😉 Proposal pointing to this PR in https://github.com/3scale/porta/pull/2293

Alternatively, if you do not like my proposal, I would prefer if you add here a 'TODO' comment for it and instead of redirecting to a legit action, just do `render head: :success, nothing: true` or even `render json: {success: true}` is an acceptable middle-step.

File: test/unit/signup/account_manager_test.rb [hunk 5]
```diff
@@ -127,7 +127,6 @@ class ProviderAccountManagerTest < Signup::AccountManagerTest
 
       test 'first service has a private endpoint in order to be functional for non-APIAP accounts' do
         Account.any_instance.stubs(:provider_can_use?).returns(false)
-        Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false)
```
Comment (Martouta) [inline test/unit/signup/account_manager_test.rb:130]: The changes in this file don't belong to this PR. And reading the title of the tests, these changes don't match what the title says either 😞

**Review by Martouta — CHANGES_REQUESTED** [review]
I've finally got to review everything in this PR except for the locales so far.

So, except for the locales that I do not know yet, everything else is good except for this:
1. https://github.com/3scale/porta/pull/2288#discussion_r537436352
2. https://github.com/3scale/porta/pull/2288#discussion_r537437896 (you do not need to do it in this same PR but then add a TODO comment)
3. https://github.com/3scale/porta/pull/2288#discussion_r537506820 (This is an important one). Check https://github.com/3scale/porta/pull/2293 pointing to this PR.
4. https://github.com/3scale/porta/pull/2288#discussion_r537509991 Also an important one. Check https://github.com/3scale/porta/pull/2294 pointing to the `master` branch.

 Btw, this is optional but recommended:
https://github.com/3scale/porta/pull/2288#discussion_r537434804

File: test/unit/signup/account_manager_test.rb (hunk 5 above)
Comment (Martouta) [reply test/unit/signup/account_manager_test.rb:130]: Proposal done in https://github.com/3scale/porta/pull/2294 so it is in a logic unit (and I added tests for related untested stuff before actually changing the code)

File: cdn/font-awesome/4.3.0/css/font-awesome.css (hunk 2 above)
Comment (josemigallas) [reply cdn/font-awesome/4.3.0/css/font-awesome.css:916]: We shouldn't even touch this files, my bad!

File: config/routes.rb [hunk 6]
```diff
@@ -12,6 +14,7 @@ def call(env)
   end
 end
 
+# rubocop:disable Metrics/BlockLength
```
Comment (Martouta) [inline config/routes.rb:17]: In this case, this comment is good, because this is just how `routes.rb` is like, and there isn't much to do about it.
👍🏼

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (Martouta) [inline config/routes.rb:1]: Plz also check the code in the whole file to ensure that we won't break anything here 😄

File: cdn/font-awesome/4.3.0/css/font-awesome.css (hunk 2 above)
Comment (Martouta) [reply cdn/font-awesome/4.3.0/css/font-awesome.css:916]: @josemigallas plz remember to remove this change then

File: features/step_definitions/proxy_steps.rb [hunk 8]
```diff
@@ -78,6 +74,7 @@ def stub_deploy_calls!
     MAPPING_RULE_ATTR.each do |attr|
       actual_value = mapping_rule.public_send(attr)
       actual_value = actual_value.name if attr == 'metric'
+      # actual_value = 'delta' if attr == 'position'
```
Comment (Martouta) [inline features/step_definitions/proxy_steps.rb:77]: We can already remove this comment, can't we?

File: config/routes.rb (hunk 7 above)
Comment (josemigallas) [reply config/routes.rb:1]: Oh, how would this break anything?

Comment (Martouta) [reply config/routes.rb:1]: this makes all string (well, not really, only literal ones) frozen, so if we make somewhere in this file something like `variable = 'hello'; variable << ' world'`, even if it is in a more subtle way, it will raise `FrozenError`now.

Comment (josemigallas) [reply config/routes.rb:1]: It looks safe, I don't see any string variable declarations or modifications.

**Review by Martouta — APPROVED** [review]
Great job 👏🏼

**Comment by hallelujah** [comment]
@josemigallas @Martouta up for merging this?
